File: src/backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/dms/curricularunit/dto/CurricularUnitDto.java
package pt.ulisboa.tecnico.rnl.dei.dms.curricularunit.dto;

import pt.ulisboa.tecnico.rnl.dei.dms.curricularunit.domain.CurricularUnit;
import pt.ulisboa.tecnico.rnl.dei.dms.person.domain.Person;

// Data Transfer Object, to communicate with frontend
public record CurricularUnitDto(long id, String code, String name, Integer semester, String course, Person mainTeacher) {
	public CurricularUnitDto(CurricularUnit curricularUnit) {
		this(
			curricularUnit.getId(), 
			curricularUnit.getCode(), 
			curricularUnit.getName(),
			curricularUnit.getSemester(),
			curricularUnit.getCourse(),
			curricularUnit.getMainTeacher()
		);
	}
}

File: src/frontend/src/models/CurricularUnitDto.ts
import type CourseDto from "./CourseDto"
import type PersonDto from "./PersonDto"

export default class CurricularUnitDto {
  id?: number
  code?: string
  name?: string
  semester?: number
  course?: CourseDto
  mainTeacher?: PersonDto

  constructor(obj?: Partial<CurricularUnitDto>) {
    Object.assign(this, obj)
  }
}